src/main/java/com/flab/yousinsa/interfaces/dto/SampleDto.java

```diff
@@ -0,0 +1,12 @@
+package com.flab.yousinsa.interfaces.dto;
+
+import com.fasterxml.jackson.annotation.JsonAutoDetect;
+
+import lombok.Data;
+
+@Data
+@JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY)
```

Comment:
이 어노테이션의 동작에 대해서 설명해주실수 있을까요?

Reply:
**문제 발생 :**

```
com.fasterxml.jackson.databind.exc.InvalidDefinitionException: No serializer found for class com.flab.yousinsa.interfaces.dto.SampleDto and no properties discovered to create BeanSerializer (to avoid exception, disable SerializationFeature.FAIL_ON_EMPTY_BEANS)
```

**해결 방안 :** 

구글링을 통해 알아본 결과, 스프링 부트에서 제공하는 Jackson 라이브러리는 Serialize하는 과정에서 접근 제한자가 public 이거나 getter/setter를 이용하기 때문에 필드가 private로 선언되어 있을 경우 JSON 변환하는 과정에서 에러가 발생한다고 합니다.

```@Data``` 어노테이션에 ```@Getter```의 기능을 가지고 있어서 문제가 발생하지 않을거라 생각했지만, 그럼에도 불구하고 에러가 발생하다보니 해당 어노테이션을 이용했습니다.

그 외에도 ```@JsonProperty``` 어노테이션을 이용하거나 ObjectMapper의 인스턴스에 private 필드에도 접근이 가능하도록 설정하는 방법이 있었는데 - 머지될 코드가 아니기도 하고 MyBatis 사용 방법 확인을 위한 샘플 코드다 보니 깊게 생각하지 않고 동작 확인만을 위해 ```@JsonAutoDetect``` 어노테이션을 적용했습니다.

Reply:
1. Lombok이 언제 getter 메소드를 추가해주나요? 
2. jackson이 getter메소드를 사용하는 시점은 언제인가요?

Reply:
1. Lombok은 컴파일 시점에 Getter 메서드가 생성됩니다.
2. Jackson이 Getter를 사용하는 시점은 Message Converter가 처리하는 과정에서 사용된다고 생각합니다.